### DISCUSSION TITLE
Can't verify contract

### DISCUSSION THREAD

**Discussion by dandexteruk** [hashlips-lab/nft-erc721-collection#448] [General]
I have deployed a contract using Truffle, and it is showing up on Etherscan, however when I try to verify the contract, I get the following error. 

yarn verify 0xC56e7cF6103e4dcF3039C0478bA7F981a2f78B44 --network truffle

yarn run v1.22.15
$ hardhat verify --constructor-args config/ContractArguments.ts 0xC56e7cF6103e4dcF3039C0478bA7F981a2f78B44 --network truffle
(node:1028) ExperimentalWarning: stream/web is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
Nothing to compile
No need to generate any newer typings.
Successfully submitted source code for contract
contracts/ChecksElite.sol:ChecksElite at 0xC56e7cF6103e4dcF3039C0478bA7F981a2f78B44
for verification on the block explorer. Waiting for verification result...

Error in plugin @nomiclabs/hardhat-etherscan: The Etherscan API responded with a failure status.
The verification may still succeed but should be checked manually.
Reason: Fail - Unable to verify. Please check if the correct constructor argument was entered.

For more info run Hardhat with --show-stack-traces
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

Any ideas, and should I at this stage be seeing the NFTs on Opensea?

Thanks

**Comment by liarco** [comment]
Hi @dandexteruk, this usually happens when you change something in your codebase between the deployment and verification process. About seeing the NFTs on OpenSea, the collection will be indexed only when there's at least one NFT minted.

If you need more help, feel free to join the [Open Devs Crew](https://opendevs.io). My team is more than happy to help holders who have questions about our tools! 😃

**Reply by dandexteruk** [reply]
Thanks for the quick reply, so does that mean I need to deploy the contract again?

**Reply by dandexteruk** [reply]
Also when I'm deploying the contract, I'm seeing this message in Truffle
![Screenshot 2023-02-23 at 14 54 58](https://user-images.githubusercontent.com/40616062/220942814-876c2523-b6ab-448b-bfd9-88fd6a6ace2f.png)
In Decoded parameters - Created contract could not be identified.
